src/UserContext.jsx

```diff
@@ -94,23 +101,21 @@ export const UserProvider = ({ children }) => {
 
 	// update localStorage values
 	const setLocalStorage = (idToken, accessToken, refreshToken) => {
-		localStorage.setItem("idToken", idToken || "null"); // Save as "null" if null
-		localStorage.setItem("accessToken", accessToken || "null");
-		localStorage.setItem("refreshToken", refreshToken || "null");
+		localStorage.setItem("KM-idToken", idToken || "null"); // Save as "null" if null
+		localStorage.setItem("KM-accessToken", accessToken || "null");
+		localStorage.setItem("KM-refreshToken", refreshToken || "null");
 	};
 
 	// Function to refresh tokens
 	const refresh = useCallback(
 		async (refreshToken) => {
-			if (refreshToken) {
-				const response = await initiateAuthRefresh(refreshToken);
-				const authResult = response.AuthenticationResult;
-				setUserDetails({
-					accessToken: authResult.AccessToken,
-					idToken: authResult.IdToken,
-					refreshToken: authResult.RefreshToken,
-				});
-			}
+			const response = await initiateAuthRefresh(refreshToken);
+			const authResult = response.AuthenticationResult;
+			setUserDetails({
+				accessToken: authResult.AccessToken,
+				idToken: authResult.IdToken,
+				refreshToken: authResult.RefreshToken,
```

Comment:
initiateAuthRefresh only returns new access and id tokens, not a new refresh token (I didn't realise that). So we don't want to overwrite the existing token.

As part of this I think we need to extend the life of refresh tokens in the cognito client config to something longer like 6 months / a year.

Reply:
ahhh, I also did not realise this. For extending the lifespan, have we outlined it in cdk already? My recollection is not and we're just using the default.

Reply:
Yeah we're just using the default of 30 days. I'm hoping we can update it in cdk and it'll update the client without us having to create a new one.

Reply:
From what I know, that should be fine. I think it only creates new IDs when you destroy and then recreate, but was that the case when you rebuilt the new pools?

Reply:
There are certain things that can't be changed once you've created the pools: I think user attributes is one. Since we're talking about something that's part of the client configuration, at worst this would only require a new client, but I'm reasonably confident this is something we can just change as and when we need to without creating a whole new client.